crypto/test/bn_test_lib.c

```diff
@@ -116,7 +116,7 @@
 #include "../bn/internal.h"
 
 
-int BN_rand(BIGNUM *rnd, int bits, int top, int bottom, RAND *rng) {
+int BN_rand(BIGNUM *rnd, int bits, RAND *rng) {
```

Comment:
This looks good.

Another option, which I prefer, would be to rewrite `TestBN2BinPadded` to be based on bn_tests.txt test vectors instead of using `BN_rand` to generate random values. e.g. Change `TestBN2BinPadded` so that it prints out its inputs and outputs in bn_test.txt format, then save the output of one or two runs, and then redo the test logic to use the bn_test.txt test vectors.

Then we could delete `BN_rand` completely.

Not required, but just letting you know what I see as being the ideal scenerio.